### PR TITLE
Include the main module header (fixes partially #4488)

### PR DESCRIPTION
**Pull request by aurianer**
This replacement is made in the examples directory only (for now)

Fixes partially #4488
i.e. replace the modules headers used in the examples by the main header of the module

### CODE REVIEW

**Comment by aurianer** [comment]
Ok so in the end this change was not scriptable, I opted for the top level headers listed in #4488, if not appropriate, for the `hpx/include/` headers and if not appropriate, either `hpx/modules/` headers. I sometimes added some extra headers to top-level or include/ headers if that made sense to me (those additions are stated in the respective commit messages), feel free to comment :)

File: components/iostreams/tests/regressions/lost_output_2236.cpp [hunk 1]
```diff
@@ -6,7 +6,7 @@
 
 #include <hpx/hpx.hpp>
 #include <hpx/hpx_init.hpp>
-#include <hpx/include/iostreams.hpp>
+#include <hpx/iostream.hpp>
```
Comment (msimberg) [inline components/iostreams/tests/regressions/lost_output_2236.cpp:9]: This we don't have yet. `hpx/iostream.hpp` should most likely be a "local" version of our current iostream support, and this particular include would have to be `hpx/distributed/iostream.hpp`. If you'd like to add the latter you can do so here.

File: examples/jacobi/jacobi_component/row_range.hpp [hunk 2]
```diff
@@ -1,10 +1,11 @@
-
 //  Copyright (c) 2012 Thomas Heller
 //
 //  SPDX-License-Identifier: BSL-1.0
 //  Distributed under the Boost Software License, Version 1.0. (See accompanying
 //  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
 
+// hpxinspect:noinclude:hpx::intrusive_ptr
```
Comment (msimberg) [inline examples/jacobi/jacobi_component/row_range.hpp:7]: I think you could change the inspect to check to expect `hpx/modules/memory.hpp` instead. Same for `row.hpp`.

File: components/iostreams/tests/regressions/lost_output_2236.cpp (hunk 1 above)
Comment (msimberg) [reply components/iostreams/tests/regressions/lost_output_2236.cpp:9]: Ah, please ignore the part about adding the file. I see you've already done that. But do change it to `hpx/distributed/iostream.hpp`.

File: examples/future_reduce/rnd_future_reduce.cpp [hunk 3]
```diff
@@ -6,9 +6,9 @@
 
 #include <hpx/hpx_init.hpp>
 #include <hpx/include/runtime.hpp>
-#include <hpx/async_combinators/when_all.hpp>
-#include <hpx/include/iostreams.hpp>
-#include <hpx/modules/timing.hpp>
+#include <hpx/include/util.hpp>
+#include <hpx/iostream.hpp>
+#include <hpx/modules/async_combinators.hpp>
```
Comment (msimberg) [inline examples/future_reduce/rnd_future_reduce.cpp:11]:
```suggestion
#include <hpx/future.hpp>
```

File: examples/1d_stencil/1d_stencil_8.cpp [hunk 4]
```diff
@@ -11,9 +11,9 @@
 
 #include <hpx/hpx_init.hpp>
 #include <hpx/hpx.hpp>
+#include <hpx/include/serialization.hpp>
```
Comment (msimberg) [inline examples/1d_stencil/1d_stencil_8.cpp:14]:
```suggestion
#include <hpx/serialization.hpp>
```

File: examples/1d_stencil/1d_stencil_7.cpp [hunk 5]
```diff
@@ -11,8 +11,8 @@
 
 #include <hpx/hpx_init.hpp>
 #include <hpx/hpx.hpp>
-#include <hpx/serialization/serialize.hpp>
-#include <hpx/type_support/unused.hpp>
+#include <hpx/include/serialization.hpp>
```
Comment (msimberg) [inline examples/1d_stencil/1d_stencil_7.cpp:14]:
```suggestion
#include <hpx/serialization.hpp>
```

File: examples/qt/widget.cpp [hunk 6]
```diff
@@ -5,7 +5,7 @@
 //  file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
 
 #include <hpx/config.hpp>
-#include <hpx/async_distributed/apply.hpp>
+#include <hpx/include/apply.hpp>
```
Comment (msimberg) [inline examples/qt/widget.cpp:8]:
```suggestion
#include <hpx/future.hpp>
```

File: libs/include/include/hpx/include/util.hpp [hunk 7]
```diff
@@ -40,3 +41,10 @@
 #include <hpx/parallel/segmented_algorithms/traits/zip_iterator.hpp>
 #include <hpx/util/activate_counters.hpp>
 #endif
+
+#if defined(HPX_HAVE_LIB_STRING_UTIL)
+#include <hpx/string_util/case_conv.hpp>
+#include <hpx/string_util/classification.hpp>
+#include <hpx/string_util/split.hpp>
+#include <hpx/string_util/trim.hpp>
```
Comment (msimberg) [inline libs/include/include/hpx/include/util.hpp:49]:
```suggestion
#include <hpx/modules/string_util.hpp>
```

File: components/iostreams/tests/regressions/lost_output_2236.cpp (hunk 1 above)
Comment (aurianer) [reply components/iostreams/tests/regressions/lost_output_2236.cpp:9]: Ok perfect will do :)

File: examples/jacobi/jacobi_component/row_range.hpp (hunk 2 above)
Comment (aurianer) [reply examples/jacobi/jacobi_component/row_range.hpp:7]: Ok I wasn't sure if we wanted to change all those includes in the main hpx files, will do :)

File: libs/include/include/hpx/include/util.hpp (hunk 7 above)
Comment (aurianer) [reply libs/include/include/hpx/include/util.hpp:49]: Thanks I thought I did that but I may have missed it ! ^^'

File: examples/jacobi/jacobi_component/row_range.hpp (hunk 2 above)
Comment (msimberg) [reply examples/jacobi/jacobi_component/row_range.hpp:7]: Don't forget to remove the `hpxinspect` comments now that they're not needed anymore.

**Comment by aurianer** [comment]
> Modulo clang formatting and removing obsolete `hpxinspect` comments this looks good. Thanks for doing this!

Done! ;)

**Comment by msimberg** [comment]
@auriane this is still left: https://208997-4455628-gh.circle-artifacts.com/0/hpx/artifacts/missing_deps.txt :/ The first missing dependency is fixed on master. The others probably come from the change to `hpx/modules/memory.hpp`. Let's merge this once that's changed.

**Pull request merged**